### ISSUE TITLE
Build fails with "Undefined symbols for architecture x86_64" under OSX

### ISSUE THREAD

**Issue by xaka** [polotek/libxmljs#315]
Hi there,

I'm getting the following error for `0.14.1` after updating from `0.10.0`:

```
Undefined symbols for architecture x86_64:
  "v8::HandleScope::CreateHandle(v8::internal::HeapObject*)", referenced from:
      libxmljs::XmlSaxParser::Callback(char const*, int, v8::Local<v8::Value>*) in xml_sax_parser.o
  "v8::HandleScope::CreateHandle(v8::internal::Object*)", referenced from:
      libxmljs::XmlAttribute::New(_xmlNode*, unsigned char const*, unsigned char const*) in xml_attribute.o
      libxmljs::XmlAttribute::New(_xmlAttr*) in xml_attribute.o
      libxmljs::XmlAttribute::get_name() in xml_attribute.o
      libxmljs::XmlAttribute::get_value() in xml_attribute.o
      libxmljs::XmlAttribute::get_namespace() in xml_attribute.o
      libxmljs::XmlAttribute::Initialize(v8::Handle<v8::Object>) in xml_attribute.o
      libxmljs::XmlDocument::Root(v8::Arguments const&) in xml_document.o
...
...
...
  "v8::Function::NewInstance() const", referenced from:
      libxmljs::XmlAttribute::New(_xmlNode*, unsigned char const*, unsigned char const*) in xml_attribute.o
      libxmljs::XmlAttribute::New(_xmlAttr*) in xml_attribute.o
      libxmljs::XmlDocument::New(_xmlDoc*) in xml_document.o
      libxmljs::XmlElement::New(_xmlNode*) in xml_element.o
      libxmljs::XmlComment::New(_xmlNode*) in xml_comment.o
      libxmljs::XmlNamespace::New(_xmlNs*) in xml_namespace.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/xmljs.node] Error 1
```

```
$ clang -v
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix

$ uname -v
Darwin Kernel Version 14.3.0: Mon Mar 23 11:59:05 PDT 2015; root:xnu-2782.20.48~5/RELEASE_X86_64
```

**Comment by defunctzombie** [comment]
I have verified that this builds on my version of osx.

```
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
💀 (~)$ uname -a
Darwin undead.local 14.3.0 Darwin Kernel Version 14.3.0: Mon Mar 23 11:59:05 PDT 2015; root:xnu-2782.20.48~5/RELEASE_X86_64 x86_64
```

Sounds like you might have something else going on in your environment.

**Comment by xaka** [comment]
What's your `npm` version? It fails on at least 2 separate and independent OSXs. I'd not call it local problem.

**Comment by defunctzombie** [comment]
2.9.0

It may be a problem but one that I am unable to confirm. As far as I can tell it works on my machine and on travis boxes.

**Comment by xaka** [comment]
I put a comment (https://github.com/polotek/libxmljs/commit/e610388f2beeb5643af0194f5712c7c2bbe2daab) so you might as well take a look. Moving away from `shared_library` target type helps.

**Comment by xaka** [comment]
In my case `npm` is `2.11.1`.

**Comment by defunctzombie** [comment]
I doubt version of npm matters for this particular issue. I am more likely to think version of node/iojs matters more.

**Comment by xaka** [comment]
I confirm it works for npm 2.9.0 and node.js 0.10.35. It breaks for latest npm 2.11.1 and latest stable node.js 0.10.38.

**Comment by xaka** [comment]
Alright. It's `node-gyp`. npm 2.9.0 uses node-gyp 1.x while npm 2.11.1 uses node-gyp 2.x, which if you look at https://github.com/TooTallNate/node-gyp/blob/master/History.md, no longer add `-undefined dynamic_lookup` to all targets, but only those, that are marked as `loadable_module`.

**Comment by defunctzombie** [comment]
sigh. ping @TooTallNate thoughts?

**Comment by othiym23** [comment]
It wasn't my intention to cause backwards-incompatible changes by updating to the new `node-gyp`, so if this is a problem, we can revert npm to use `node-gyp@1` and roll the new version out in a subsequent semver-major version of npm (`npm@3` may be too soon, so it might have to wait until `npm@4`).

**Comment by xaka** [comment]
The fix seems trivial for this particular project and I haven't been able to find any other similar issues so far, so I'd not say we need a rollback, it should be 2nd option IMHO.

**Comment by xaka** [comment]
My actual concern is around marking binding module (not it dependencies) as `shared_library`, which is unusual AFAIK based on my research and causing the problem, but I know so little about `node-gyp` and developing bindings that I'd prefer somebody else throw ideas here.

**Comment by defunctzombie** [comment]
@xaka I don't think we can remove the line in the bindings file since it was added to address the issues described here:

https://github.com/polotek/libxmljs/pull/247

If those can be solved some other cross npm/gyp compatible way I am all ears. It is getting increasingly difficult (bordering on impossible) to support all the various cross combinations of npm/node/iojs updates because the gyp file itself isn't a makefile I can just run selectively.

**Comment by defunctzombie** [comment]
ping @albanm since you did the work on the lines in question here

**Comment by xaka** [comment]
Doesn't look like we're going anywhere with this. Do we want to keep `shared_library`, but add `-undefined dynamic_lookup` to linker for now as a workaround / temporary solution?

**Comment by defunctzombie** [comment]
@xaka yea, if that will fix it and allow for the shared_library to remain that sounds like the ideal solution to me.

**Comment by Splurov** [comment]
https://github.com/polotek/libxmljs/commit/c95f43b70410675dc3ae473369387e3275e7594c fixed the exact same issue for me (npm 2.11.3, node 0.12.5).

**Comment by danielb2** [comment]
:+1: Also have errors compiling.

**Comment by defunctzombie** [comment]
This has been fixed in 0.14.2
